flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -257,7 +257,9 @@ object CodeGenUtils {
     case TIMESTAMP_WITHOUT_TIME_ZONE | TIMESTAMP_WITH_LOCAL_TIME_ZONE =>
       s"$term.hashCode()"
     case TIMESTAMP_WITH_TIME_ZONE | ARRAY | MULTISET | MAP =>
-      throw new UnsupportedOperationException("Unsupported type: " + t)
+      throw new UnsupportedOperationException(
+        s"Unsupported type($t) to generate hash code," +
+            s" the type($t) is not supported as a GROUP_BY/PARTITION_BY/JOIN_EQUAL field.")
```

Comment:
Also not supported in UNION, right?

Reply:
You mean UNION distinct?
Actually including all set distinct operators, most of them are not supported in Flink batch/stream. I think we don't need list here.